shell/platform/tizen/accessibility_settings.cc

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "accessibility_settings.h"
+#include "flutter/shell/platform/tizen/flutter_tizen_engine.h"
+#include "flutter/shell/platform/tizen/logger.h"
+
+#if defined(TV_PROFILE) && (!defined(__X64_SHELL__))
+#include <system/system_settings.h>
+#endif
+
+// SYSTEM_SETTINGS_KEY_MENU_SYSTEM_ACCESSIBILITY_HIGHCONTRAST = 10059 has been
+// defined in system_settings_keys.h only for TV profile
+#define SYSTEM_SETTINGS_KEY_MENU_SYSTEM_ACCESSIBILITY_HIGHCONTRAST 10059
+
+namespace flutter {
+#if defined(TV_PROFILE) && (!defined(__X64_SHELL__))
+static void OnAccessibilityHighContrastChangedCallback(
+    system_settings_key_e key,
+    void* user_data) {
+  auto accessibilitySettings =
+      reinterpret_cast<AccessibilitySettings*>(user_data);
+  accessibilitySettings->OnAccessibilityHighContrastStateChanged();
+}
+#endif
+
+AccessibilitySettings::AccessibilitySettings(FlutterTizenEngine* engine)
+    : engine_(engine) {
+  Init();
+}
+
+AccessibilitySettings::~AccessibilitySettings() {
+  Dispose();
+}
+
+void AccessibilitySettings::OnAccessibilityHighContrastStateChanged() {
+  if (engine_ != nullptr) {
+    engine_->SetAccessibilityHighContrastEnabled(
+        GetAccessibilityHighContrastValue());
+  }
+}
+
+void AccessibilitySettings::Init() {
+#if defined(TV_PROFILE) && (!defined(__X64_SHELL__))
+  // Add listener for accessibility high contrast
+  system_settings_set_changed_cb(
+      system_settings_key_e(
+          SYSTEM_SETTINGS_KEY_MENU_SYSTEM_ACCESSIBILITY_HIGHCONTRAST),
+      OnAccessibilityHighContrastChangedCallback, this);
+
+  // Set initialized value of accessibility high contrast
+  if (engine_ != nullptr) {
+    engine_->SetAccessibilityHighContrastEnabled(
+        GetAccessibilityHighContrastValue());
+  }
+#endif
+}
+
+void AccessibilitySettings::Dispose() {
+#if defined(TV_PROFILE) && (!defined(__X64_SHELL__))
+  system_settings_unset_changed_cb(system_settings_key_e(
+      SYSTEM_SETTINGS_KEY_MENU_SYSTEM_ACCESSIBILITY_HIGHCONTRAST));
+#endif
+}
+
+int AccessibilitySettings::GetAccessibilityHighContrastValue() {
+  int enabled = 0;
+#if defined(TV_PROFILE) && (!defined(__X64_SHELL__))
+  int ret = system_settings_get_value_int(
+      system_settings_key_e(
+          SYSTEM_SETTINGS_KEY_MENU_SYSTEM_ACCESSIBILITY_HIGHCONTRAST),
+      &enabled);
+  if (ret != SYSTEM_SETTINGS_ERROR_NONE) {
+    FT_LOG(Error)
+        << "Failed to get value of accessibility high contrast. ERROR CODE = "
+        << ret;
+  }
+#endif
+  return enabled;
```

Comment:
> You might simply change the return type to `bool`.
> 
> By the way cannot `system_settings_set_value_bool` be used instead of `system_settings_get_value_int`?

I had tried system_settings_set_value_bool, but met the parameter error

Reply:
done

Reply:
> Looks much better than before. Thanks!

Thanks for your review. I have modified all of them.
 Please check.